### ISSUE TITLE
OpenCms 10.5: CmsSolrIndex#getDocument vs. CmsSearchIndex#getDocument

### ISSUE THREAD

**Issue by kartobi** [alkacon/opencms-core#480]
org.opencms.search.CmsSolrIndex#getDocument(java.lang.String, java.lang.String) overwrites org.opencms.search.CmsSearchIndex#getDocument(java.lang.String, java.lang.String) but returns null if several results are found.
```                
if ((sdl.getNumFound() == 1L) && (sdl.get(0) != null)) {
    return new CmsSolrDocument(sdl.get(0));
}
```
(line 355)

vs.
```
if (hits.scoreDocs.length > 0) {
    result = searcher.doc(hits.scoreDocs[0].doc);
}
```
(line 728)

Is this different behaviour by purpose?
If so, why did you change it?

Best regards,
Tobias

**Comment by dSeidel** [comment]
Hi @kartobi,
there seems to be no reason for the different behavior of CmsSolrIndex#getDocument(String, String). Internally, the method is just used to get documents by structure-id or path, thus there should always be just a single result and the difference has no effect.
I will change the behavior to be similar to CmsSearchIndex#getDocument(String,String). This will be part of Release 10.5.1.

**Comment by tHerrmann** [comment]
This was fixed with release 10.5.1.
